### ISSUE TITLE
Error in htmldomapi file

### ISSUE THREAD

**Issue by carloslfu** [paldepind/snabbdom#110]
Hi @paldepind , I have updated my code to the last version of snabbdom and dipatch this error in the browser: `htmldomapi.js:15 Uncaught NotFoundError: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.`

I have isolated the error and this the minimum code that reproduces the same issue:

``` javascript
const patch = require('snabbdom').init([])
const h = require('snabbdom/h')

let container = document.getElementById('app')

let vnode0 = h('div', {key: '0'}, [
  h('div.class0', {key: '0.0'}, '0.0')
])

let vnode1 = h('div', {key: '0'}, [
  h('div.class1', {key: '0.0'}, '0.0')
])

let vnode = patch(container, vnode0)

setTimeout(function() {
  patch(vnode, vnode1)
}, 2000)

```

**Comment by paldepind** [comment]
Thank you for reporting this.

Is the `setTimeout` necessary to trigger the issue?

**Comment by yuanhan1890** [comment]
Indeed, you misused **snabbdom**.

**Snabbdom** patchs vnodes before checking the equality of `vnode.sel` and `vnode.key`, and when to patch the children of vnodes, **snabbdom** will probably  fallback to `key`. For more details, you can check the implementation of [updateChildren](https://github.com/paldepind/snabbdom/blob/master/snabbdom.js#L133)

When you patch the child `h('div.class1', {key: '0.0'}, '0.0')` to `h('div.class0', {key: '0.0'}, '0.0')`, **snabbdom** will [check](https://github.com/paldepind/snabbdom/blob/master/snabbdom.js#L148) the `sel ` of `childVnode` firstly and then turn to [the key comparasion](https://github.com/paldepind/snabbdom/blob/master/snabbdom.js#L166).

During the key comparasion, `elmToMove` is equal to `oldStartVnode`. **snabbdom** removes `elmToMove.elm` from `parentElement`, and then use `oldStartVnode.elm` which is indeed `elmToMove.elm` and for now `elmToMove.elm.parentElement` equals `null` in `insertBefore` function, resulting errors.

**To sum up, When to use `key` attribute, it's better to use the  same `sel` attribute.**

ps: I‘m unaccustomed to write in english. If you find my answer helps, check my grammer, tell me and i will appreciate it ^_^

**Comment by carloslfu** [comment]
@paldepind `setTimeout` isn't necesary. @yuanhan1890 thanks for the explanation, it's very useful.

**Comment by caesarsol** [comment]
Still, I think this is a snabbdom bug... The different `sel` should prevent the patch and do a node replacement instead. Or the `key` should prevail and update the selector. Anyways it should not cause an error, in my opinion.

Is this still happening on master?

As a side note, the correct use of `key` should be mentioned on the README.
